elements/map/src/plugins/globe/methods.js

```diff
@@ -0,0 +1,157 @@
+import { createXYZ } from "ol/tilegrid";
+import { getFlatLayersArray } from "../../helpers/layer.js";
+
+/** @typedef {import("./types").ExtendedOLLayer} ExtendedOLLayer */
+
+export const createMapPool = (maxMaps, EOxMap, target) => {
+  // Get the serializable definitions of all layers on the main map.
+  const allMainLayers = getFlatLayersArray(EOxMap.map.getLayers().getArray());
+  const layerDefs = allMainLayers
+    .map((l) => l.get("_jsonDefinition"))
+    .filter(Boolean);
+
+  return Array.from({ length: maxMaps }, () => {
+    const tileMap = /** @type {import("../../main").EOxMap} **/ (
+      document.createElement("eox-map")
+    );
+    Object.assign(tileMap.style, {
+      width: `256px`,
+      height: `256px`,
+      position: "absolute",
+      top: `-9999px`,
+      left: `-9999px`,
+    });
+
+    // Initialize the tileMap with its own instances of all layers
+    Object.assign(tileMap, {
+      projection: "EPSG:3857",
+      layers: layerDefs,
+    });
+
+    target.appendChild(tileMap);
+    return {
+      tileMap,
+      tileQueue: [],
+      loadNextTile() {
+        const job = this.tileQueue[0];
+        requestTileFromMap(tileMap, job.layerId, job, this);
+      },
+    };
+  });
+};
+
+async function requestTileFromMap(tileMap, layerId, job, mapPoolMap) {
+  const map = tileMap.map;
+
+  // Wait for the map to be mounted before proceeding
+  if (!map) {
+    tileMap.addEventListener(
+      "mapmounted",
+      () => requestTileFromMap(tileMap, layerId, job, mapPoolMap),
+      { once: true },
+    );
+    return;
+  }
+
+  // Instead of re-creating the layer, make sure only the target layer is visible
+  map.getLayers().forEach((l) => {
+    l.setVisible(l.get("id") === layerId);
+  });
+
+  const tileGrid = createXYZ();
+  const extent = tileGrid.getTileCoordExtent([
+    job.tile.z,
+    job.tile.x,
+    job.tile.y,
+  ]);
+
+  map.getView().fit(extent, {
+    callback: () => {
+      map.once("rendercomplete", function () {
+        const mapCanvas = document.createElement("canvas");
+        const size = map.getSize();
```

Comment:
in the latest version of `ol`, getting the imageData of the mapCanvas should be as easy as `mapCanvas.transferToImageBitmap()`